libsvm-format/java-formats/libsvm/src/com/dataiku/dss/formats/libsvm/LIBSVMFormat.java

```diff
@@ -0,0 +1,261 @@
+        throw new UnsupportedOperationException("No output for this format.");
+    }
+    
+    /**
+     * Create a schema detector for a stream in the format (used if canReadSchema=true in the json)
+     */
+    @Override
+    public CustomFormatSchemaDetector getDetector(JsonObject config, JsonObject pluginConfig) {
+        throw new UnsupportedOperationException("No detector for this format.");
+    }
+
+    public class LIBSVMFormatInput implements CustomFormatInput {
+        /**
+         * Called if the schema is available (ie, dataset has been created)
+         */
+        @Override
+        public void setSchema(Schema schema, boolean allowExtraColumns) {
+        }
+
+        @Override
+        public void setWarningsContext(WarningsContext warnContext) {
+        }
+        
+        private void parseIntoMultiColumn(ProcessorOutput out, ColumnFactory cf, RowFactory rf,
+                                          BufferedReader bf, LimitedLogContext limitedLogger) throws Exception {
+            HashMap<String, Column> columns = new HashMap<>();
+            Column labelColumn = cf.column("Label");
+
+            Matcher labelMatcher = LABEL_PATTERN.matcher("");
+            Matcher tokenMatcher = TOKEN_PATTERN.matcher("");
+
+            String line;
+            int lineNumber = 0;
+
+            while ((line = bf.readLine()) != null) {
+                lineNumber++;
+                labelMatcher.reset(line);
+                tokenMatcher.reset(line);
+
+                // If no label is found the line is considered as invalid and skipped
+                if (!labelMatcher.find()) {
+                    limitedLogger.logV("Line %d is invalid (no label detected): %s", lineNumber, line);
+                    continue;
+                }
+                Row row = rf.row();
+
+                String label = labelMatcher.group(1);
+                row.put(labelColumn, label);
+
+                int start = labelMatcher.end();
+                tokenMatcher.region(start, line.length());
+
+                while (tokenMatcher.find()) {
+                    String index = tokenMatcher.group(1);
+                    String value = tokenMatcher.group(2);
+
+                    Column column = columns.get(index);
+                    // Creating the newly found column if possible
+                    if (column == null) {
+                        if (columns.size() > maxFeatures) continue;
+
+                        column = cf.column(index);
+                        columns.put(index, column);
+                    }
+
+                    start = tokenMatcher.end();
+                    tokenMatcher.region(start, line.length());
+
+                    row.put(column, value);
+                }
+
+                if (!tokenMatcher.hitEnd()) {
+                    limitedLogger.logV("Invalid end-of-line at [%d,%d]: %s", lineNumber, start, line.substring(start));
```

Comment:
The _token_ is invalid, not the end of the line (+ can be confusing if the error is around the middle, some users won't read the column position) → something like "Invalid token at line %d, column %d: '%s', ignoring the rest of that line".